pom.xml

```diff
@@ -356,8 +356,24 @@
                 <systemPropertyVariables>
                   <jacoco-agent.destfile>${jacoco.exec.file}</jacoco-agent.destfile>
                 </systemPropertyVariables>
+                <skip>${code.coverage.disabled}</skip>
               </configuration>
             </plugin>
+            <plugin>
+              <artifactId>maven-compiler-plugin</artifactId>
```

Comment:
Would you please explain why skipping test compilation is necessary?

Reply:
@rsynek 
That profile is used only for sonar reports. So, all the time spent on "testCompile" modules that are ignored by jacoco and sonar is wasted.

Reply:
I think that's a false assumption - the sonarcloud analysis is bound to `generate-resources` phase, which is called from jenkins jobs:
https://github.com/kiegroup/kie-jenkins-scripts/blob/master/job-dsls/jobs/pr_jobs.groovy#L254

It's not ideal; I should have made it even more early - `validate`. So if you want to optimize the execution time even further, I would recommend binding the maven execution on Jenkins to the `validate` phase and of course making a corresponding change in kie-parent.

The described optimization is out of scope of this PR, though.

Reply:
@rsynek 
I've see that while launching commands, that's why I've put that that way.
Please verify yourself.

Reply:
It depends on what command did you use to run the analysis. If I run it via:
`mvn generate-resources -Psonarcloud-analysis`, there is no invocation of the compiler plugin.

Reply:
@rsynek 
That exclusion is inside the run-code-coverage profile, not in the sonarcloud one.

Reply:
I see. So you want to skip compilation of tests in the original maven goal that is supposed to run tests? This leads me to a question - are these tests from modules where you skip their compilation executed?